test/param_build_test.c

```diff
@@ -83,13 +87,21 @@ static int template_public_test(void)
         /* Check UTF8 pointer */
         || !TEST_ptr(p = OSSL_PARAM_locate(params, "utf8_p"))
         || !TEST_true(OSSL_PARAM_get_utf8_ptr(p, &cutf))
-        || !TEST_str_eq(cutf, "bar-boom"))
+        || !TEST_str_eq(cutf, "bar-boom")
+        /* Check BN */
+        || !TEST_ptr(p = OSSL_PARAM_locate(params, "bignumber"))
+        || !TEST_str_eq(p->key, "bignumber")
+        || !TEST_uint_eq(p->data_type, OSSL_PARAM_UNSIGNED_INTEGER)
+        || !TEST_true(OSSL_PARAM_get_BN(p, &bn_res))
+        || !TEST_int_eq(BN_cmp(bn_res, bn), 0))
         goto err;
     res = 1;
 err:
     OPENSSL_free(params);
     OPENSSL_secure_free(secure);
```

Comment:
Considering that `OPENSSL_secure_free()` doesn't always clear the memory chunk, I would say it's categorically *wrong* not to use `OPENSSL_secure_clear_free()` for the supposedly secure memory chunk.

Reply:
On the other hand, since you have already tested that `secure` is NULL, this call is pointless and should probably be removed.  We're not testing `OPENSSL_secure_free()` here!

Reply:
The tests are presumed to be example code, thus freeing the pointer is good practice.
Not clearing it at the same time is bad practice.

Reply:
I'll accept the former complaint.
The latter not since the tests are supposed to be example code...

Reply:
What happens if the test is failing and some secure memory was allocated -- that needs to be freed.

Reply:
What happens if code fails and the function as allocated resources?  It frees them.  Why is this different?